apache/knox
# KNOX-2834 - Bumped angular/cli to 14.2.9 in admin-ui, token-generation-ui, homepage-ui, token-management-ui, webshell-ui

## What changes were proposed in this pull request?

The `angular/cli` dependency of all angular applications have been upgraded to `14.2.9`. Since the projects have been on `1.7.4` (except webshell-ui),  other dependencies, files, configs and some code needed to be changed too.
Main changes:
- changed `TSLint` to `ESLint`
- switched `angular-cli.json` to `angular.json`
- some files were not linted, they are now linted and formatted
- `sweetalert` is changed to `sweetalert2`
- nodejs version is now `v14.15.0`
- commented out xml parsing of shared providers in `resource-detail.component`
- the div that hosts the topology `ace-editor` is now hidden (it was not generated before) until it is needed, so that 
it's `ViewChild` can be set.

## How was this patch tested?

I manually tested the features:

Token-management:
- generated normal token
- generated impersonated token
- disabled impersonation
- set server-managed to false
- disabled and revoked the created tokens

Token-generation:
- set server-managed to true and false
- set impersonation to true and false
- set lifespan.input.enabled to true and false
- tried to generate token over the max lifespan limit
- generated tokens until reached token limit
- authenticated with a generated token

Homepage:
- Tried the links
- Mouse on service name -> description shows up
- When UI services are clicked they navigate to UI URL
- When API services are clicked description shows up
- Dropdown menus open and close
- New topologies appear under the topologies menu 

Admin-ui:
- Created, deleted, modified service definitions
- Created, deleted, modified provider configuration
- Created, deleted, modified descriptor
- Copied, deleted, modified topology

WebShell-UI:
- Configured webshell and link appeared on homepage
- Signed in to knox with the user tom (who is a user on my local computer) and was able to connect to my computer
